tests/test_flows.py

```diff
@@ -401,18 +401,88 @@ def test_channel_selection(self):
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
 
-        self.debug("Send channel selection request")
-        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x8006)
-        time.sleep(1)
-        self.debug("Confirming channel selection request has been received on agent")
+        # Send empty channel selection request
+        self.debug("Send empty channel selection request")
+        self.gateway_ucc.dev_send_1905(self.mac_repeater1,0x8006)
+
+        self.debug("Confirming empty channel selection request has been received on agent")
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
 
-        #self.debug("Confirming 1905.1 Ack Message request was received on agent")
-        # TODO: When creating handler for the ACK message on the agent, replace lookup of this string
-        # TODO: currently controller sends empty channel selection request, so no switch is performed
-        # self.check_log(self.repeater1, "agent_wlan0", "ACK_MESSAGE")
-        # self.check_log(self.repeater1, "agent_wlan2", "ACK_MESSAGE")
+        self.debug("Confirming OPERATING_CHANNEL_REPORT_MESSAGE message has been received on controller")
+        self.check_log(self.gateway, "controller", "OPERATING_CHANNEL_REPORT_MESSAGE")
+
+        # Send channel selection request with chnging tx_power_limit 
+        # Payload for transmit power - 20 for both radio
+        payload_transmit_power ="0x14"
+
+        for i in range(1, 3):
+            self.debug("Send empty channel selection request with changing tx_power_limit")
+            self.gateway_ucc.dev_send_1905(
+                self.mac_repeater1,
+                0x8006,
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan0} {payload_transmit_power}'),
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan2} {payload_transmit_power}')
+            )
+
+            self.check_log(self.repeater1, "agent_wlan0", "tlvTransmitPowerLimit %d" % int(payload_transmit_power,16))
+            self.check_log(self.repeater1, "agent_wlan2", "tlvTransmitPowerLimit %d" % int(payload_transmit_power,16))
+            self.check_log(self.gateway, "controller", "tx_power=%d" % int(payload_transmit_power,16))
```

Comment:
Maybe keep result of `int(payload_transmit_power,16)` in local variable and avoid calculation duplication?